### PR TITLE
[Bugfix][TPU] Do not use torch.Generator for TPUs

### PR DESCRIPTION
**Pull request by WoosukKwon**
Currently, load_format="dummy" is broken on TPUs because TPU does not support `torch.Generator`. Since it is not always necessary to make the dummy weights deterministic, we can directly call `.uniform_`, not using the seed.

### CODE REVIEW

File: vllm/model_executor/model_loader/weight_utils.py [hunk 1]
```diff
@@ -490,6 +491,11 @@ def initialize_dummy_weights(
     """
     for param in model.state_dict().values():
         if torch.is_floating_point(param):
+            if current_platform.is_tpu():
+                # XLA device does not support torch.Generator()
+                param.uniform_(low, high)
```
Comment (youkaichao) [inline vllm/model_executor/model_loader/weight_utils.py:496]: how does xla support seed then?

Comment (WoosukKwon) [reply vllm/model_executor/model_loader/weight_utils.py:496]: They have an API, which we use in `init_device`: https://github.com/vllm-project/vllm/blob/533d1932d23917f7a29ba04c6be5c8b2676b6969/vllm/worker/tpu_worker.py#L97

Comment (youkaichao) [reply vllm/model_executor/model_loader/weight_utils.py:496]: Does it support per-function seed?

Comment (WoosukKwon) [reply vllm/model_executor/model_loader/weight_utils.py:496]: Not at the moment according to the pytorch xla team.

**Review by youkaichao — APPROVED** [review]
please fix the lint error
